src/main/java/Reporter.java

```diff
@@ -27,65 +28,99 @@
-          fut1.complete();
+          fut0.complete();
         } else {
-          fut1.fail("Fail when Creating feed");
+          fut0.fail("Fail when creating feed");
         }
       });
-      JsonObject json = new JsonObject().put("id", feedId);
-      request.end(json.encode());
-
-      fut1.compose(response -> {
-        // step 2: create a vertx
-        HttpClientRequest request1 = reporter.getHttpRequest(host, resourceURI, HttpMethod.POST).handler(response1 -> {
-          System.out.println("Created vertx resource!");
-          fut2.complete();
+      request.end(new JsonObject().put("id", feedId).encode());
+
+      fut0.compose(Void -> {
+        // step 1: create customized resource type
+        HttpClientRequest req = reporter.getHttpRequest(host, resourceTypeURI, HttpMethod.POST).handler(resp -> {
+          if (resp.statusCode() == 201) {
+            System.out.println("Created customized resource type");
+            fut1.complete();
+          } else {
+            fut1.fail("Fail when creating resource type");
+          }
         });
-        JsonObject json1 = new JsonObject().put("id", vertxResourceId).put("resourceTypePath", "/rt;URL")
+        JsonObject json = new JsonObject().put("id", "MYRT");
+        req.end(json.encode());
+      }, fut1);
+
+      fut1.compose(Void -> {
+        // step 2: create a vertx resource
+        HttpClientRequest req = reporter.getHttpRequest(host, resourceURI, HttpMethod.POST).handler(resp -> {
+          if (resp.statusCode() == 201) {
+            System.out.println("Created vertx resource!");
+            fut2.complete();
+          } else {
+            fut2.fail("Fail when creating resource vertx");
+          }
+        });
+        JsonObject json = new JsonObject().put("id", vertxResourceId).put("resourceTypePath", "/f;" + feedId + "/rt;MYRT")
                 .put("properties", new JsonObject().put("type", "standalone"));
-        request1.end(json1.encode());
+        req.end(json.encode());
       }, fut2);
 
-      fut2.compose(response -> {
+      fut2.compose(Void -> {
         // step 3: create a event bus resource
-        HttpClientRequest request2 = reporter.getHttpRequest(host, resourceURI, HttpMethod.POST).handler(response2 -> {
-          System.out.println("Created event_bus resource!");
-          fut3.complete();
+        HttpClientRequest req = reporter.getHttpRequest(host, resourceURI, HttpMethod.POST).handler(resp -> {
+          if (resp.statusCode() == 201) {
+            System.out.println("Created event_bus resource!");
+            fut3.complete();
+          } else {
+            fut3.fail("Fail when creating resource event bus");
+          }
         });
-        JsonObject json2 = new JsonObject().put("id", eventBusResourceId).put("resourceTypePath", "/rt;URL");
-        request2.end(json2.encode());
+        JsonObject json = new JsonObject().put("id", eventBusResourceId).put("resourceTypePath", "/f;" + feedId + "/rt;MYRT");
```

Comment:
Does this mean that the resource type is the same for both the Vert.x resource and the EventBus resource?